cmake/libdeflate.pc.in

```diff
@@ -0,0 +1,10 @@
+prefix=@CMAKE_INSTALL_PREFIX@
+exec_prefix=${prefix}
+includedir=${prefix}/@CMAKE_INSTALL_INCLUDEDIR@
+libdir=${exec_prefix}/@CMAKE_INSTALL_LIBDIR@
```

Comment:
The Makefile did the equivalent of specifying these all directly as `CMAKE_INSTALL_FULL_*`, which is technically a bad idea (pkg-config likes to have paths be prefix-based, there are cases where it matters) but generally worked okay-ish.

This however is actually broken. If `CMAKE_INSTALL_*` is the same as `CMAKE_INSTALL_FULL_*`, i.e. the paths specified on the cmake command line are `-DCMAKE_INSTALL_LIBDIR=/usr/lib` instead of `-DCMAKE_INSTALL_LIBDIR=lib`, the final path in the pkg-config file will resolve as `libdir=/usr/usr/lib`.

The Meson build system solves this problem in two ways:
- the correctness guarantee of the built-in `pkgconfig.generate()`, which always specifies these variables correctly and merely expects a project to declare which library targets are being exported into the pkg-config file
- you can write your own functionality:
  - relying on the builtin `join_paths()` which correctly handles prepending the string `${prefix}` to the libdir, but only if libdir isn't an absolute path
  - also the equivalent of `CMAKE_INSTALL_LIBDIR` is `get_option('libdir')` and this guarantees that even when the user specifies an absolute path on the command line, if it begins with the prefix then the value will be normalized to the relative version. As a result, you won't ever end up with `libdir=/usr/lib` when `libdir=${prefix}/lib` is preferred.

Emulating any of this in cmake is complicated. You cannot handle point 2b, point 2a requires reimplementing a `join_paths()`-like function, option 3 is to give up and just hardcode `CMAKE_INSTALL_FULL_*` everywhere.

Reply:
What would be the use case of an absolute path in `CMAKE_INSTALL_LIBDIR`? Is there any project doing that? I mean, it's allowed unfortunately but discouraged.

Reply:
If you have a robust implementation handling all these edge cases, please share :)

Reply:
Even more unfortunately, it's not actually about the project. The project doesn't specify any of the paths. The random user trying to build it, is the one that specifies the paths -- and random users are likely to do all sorts of unexpected things. :)

Reply:
https://github.com/jtojnar/cmake-snips#concatenating-paths-when-building-pkg-config-files

A `join_paths()` function that correctly handles point 2a, and treats point 2b as out of scope ("if you the user specify absolute paths, then you the user will end up with inflexible pkg-config files that may not correctly handle cross compilation sysroots and other less common use cases").


It would be nice if cmake internally normalized the non `*_FULL*` variables.

Reply:
No but project could catch the values passed externally and deduce the proper values to inject in pkgconfig file. Yet, is there any CMake project out there which handle this edge case?

Reply:
Thanks for the tips, I'll try this function.